### PR TITLE
Formula runner API serializables.

### PR DESCRIPTION
**Pull request by heavenlyhash**
`repeatr//api/act` now has some more sub-packages that implement a server and a client for passing the serialized log objects.

The `def.SerializedOutput` type is now known simply as `def.Event`, but serial forms are unchanged.

There's quite a few things still filed as "future work", but the client side should fly right.  (The server proxying side is quite a bit of work to detangle, because the log objects and all the other writers have never before been unified... but what we do emit from stdout should in fact parse through this client.)

### CODE REVIEW

File: api/act/remote/runObserver.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		evt := roc.readOne()
+		if evt == (def.Event{}) {
+			break
+		}
+		stream <- &evt
+	}
+}
+
+func (roc *RunObserverClient) AwaitRunRecord(def.RunID) *def.RunRecord {
+	return nil // TODO
```
Comment (ericevenchick) [inline api/act/remote/runObserver.go]: would probably prefer a panic of type ErrUnimplemented or something

File: api/act/remote/runObserver.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	// It's something you can generate by watching the stream;
+	// it makes no sense to have every stream implementation watch for it itself.
+}
+
+func (roc *RunObserverClient) readOne() def.Event {
+	out := def.Event{}
+
+	err := codec.NewDecoder(roc.Remote, roc.Codec).Decode(&out)
+	meep.TryPlan{
+		{ByVal: io.EOF,
```
Comment (ericevenchick) [inline api/act/remote/runObserver.go]: why discard EOFs?

File: api/act/running.go [hunk 3]
```diff
@@ -3,10 +3,69 @@ package act
...
+	*/
+	AwaitRunRecord(def.RunID) *def.RunRecord
+}
+
+type ErrRemotePanic struct {
+	meep.TraitAutodescribing
+	Dump string
+}
+
+type ErrNotMyProblem struct {
```
Comment (ericevenchick) [inline api/act/running.go]: I would like docs because this error seems somewhat underspecified

File: api/act/remote/runObserver.go (hunk 1 above)
Comment (heavenlyhash) [inline api/act/remote/runObserver.go]: Now nuked in latest, per the reasoning in that "REVIEW" comment.

File: api/act/remote/runObserver.go (hunk 2 above)
Comment (heavenlyhash) [inline api/act/remote/runObserver.go]: Refactored to actually return a clear state flag.  (Same behavior as before; less weird looking and less abuse of the zero-val of large structs as a sentinel val.)

File: api/act/running.go (hunk 3 above)
Comment (heavenlyhash) [inline api/act/running.go]: Renamed to be less bad.

**Comment by heavenlyhash** [comment]
Added thing to gather dumps on nonsense messages, too.  Pretty happy with this now.

**Pull request merged**